org.eclipse.xtext/src/org/eclipse/xtext/mwe/PathTraverser.java

```diff
@@ -52,22 +67,18 @@ public Set<URI> findAllResourceUris(String path, Predicate<URI> isValidPredicate
 		throw new IllegalArgumentException("Unsupported path : " + path + " (only folders and archives are supported).");
 	}
 
+	/** @deprecated Instead use {@link #traverseArchive(File, Predicate)} */
+	@Deprecated(since = "2.35.0", forRemoval = true)
+	protected Set<URI> traverseArchive(File file, com.google.common.base.Predicate<URI> isValidPredicate) {
```

Comment:
I wonder if it is necessary to keep these protected methods as deprecated?
Are custom sub-classes of `PathTraverser` something that should be supported?
If yes, it should probably also be ensured that these deprecated methods are still called in order to not make custom implementation dysfunctional.

The same applies for `Reader` and `RuntimeResourceSetInitializer` adjusted in this PR.